pkg/edvprovider/couchdbedvprovider/couchdbedvprovider.go

```diff
@@ -63,14 +66,40 @@ func NewProvider(databaseURL, dbPrefix string) (*CouchDBEDVProvider, error) {
 	return &CouchDBEDVProvider{coreProvider: couchDBProvider}, nil
 }
 
-// CreateStore creates a new store with the given name.
+// CreateStore creates a new store. If the given name is a base58-encoded 128-bit value, we decode and creates a uuid
```

Comment:
I allowed regular strings to pass too because the config store name is not encoded. Does it make more sense to throw an error here for everything except the config store name?

Reply:
Ah, I forgot about that. In that case, I don't have a strong preference. It seems ok to me as is, but perhaps another reviewer will have a preference.

Reply:
Although you might consider still using that method I referenced earlier so as to prevent any weird potential edge-cases and to prevent duplication of code.

e.g. is it possible to base58 decode a regular name and accidentally get something that's 16 bytes long? i.e. a false positive of sorts?

Reply:
(you could refactor that checkIfBase58Encoded128BitValue() function to work better for your use case if need be)

Reply:
done